js/main.js

```diff
@@ -145,23 +153,36 @@ function handleClick(event) {
     data.view = 'entry-form';
     $h2EditEntryTitle.className = '';
     $h2NewEntriesTitle.className = 'hidden';
-
+    deleteEntry.className = 'delete-entry';
+  } else if (event.target.matches('.delete-entry') === true) {
+    modalBox.className = 'modal-box';
+    overlay.className = 'overlay';
+  } else if (event.target.matches('.modal-cancel-color') === true) {
+    modalBox.className = 'hidden';
+    overlay.className = 'hidden';
+  } else if (event.target.matches('.modal-confirm-color') === true) {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.editing.entry === data.entries[i].entry) {
+        var $allLi = document.querySelectorAll('li');
+        data.entries.splice([data.entries[i]], 1);
```

Comment:
Does this work? The first argument of `splice` is meant to be the index to start removing from 🤔